plugins/Marketplace/Controller.php

```diff
@@ -305,6 +274,34 @@ public function overview()
         return $view->render();
     }
 
+    public function searchPlugins(): string
+    {
+        Piwik::checkUserIsNotAnonymous();
+
+        $request = Request::fromRequest();
+
+        $query = $request->getStringParameter('query', '');
+        $themesOnly = $request->getBoolParameter('themesOnly', false);
+        $purchaseType = $request->getStringParameter('purchaseType', '');
+        $sort = $request->getStringParameter('sort', '');
+
+        $purchaseType = (new PurchaseType())->getPurchaseType($purchaseType);
+        $sort = (new Sort())->getSort($sort);
+
+        $plugins = $this->plugins->searchPlugins($query, $sort, $themesOnly, $purchaseType);
+
+        foreach ($plugins as &$plugin) {
+            if (!$plugin['isDownloadable']) {
+                continue;
+            }
+
+            $plugin['downloadNonce'] = Nonce::getNonce(static::DOWNLOAD_NONCE_PREFIX . $plugin['name']);
```

Comment:
Positive-logic conditions are generally easier to understand and the loop is less side-effect-y..

```suggestion
            if ($plugin['isDownloadable']) {
                $plugin['downloadNonce'] = Nonce::getNonce(static::DOWNLOAD_NONCE_PREFIX . $plugin['name']);
            }
```